assets/src/components/selection/datetime-select/index.js

```diff
@@ -0,0 +1,228 @@
+/**
+ * External dependencies
+ */
+import { stringify } from 'querystringify';
+import moment from 'moment';
+import { isUndefined, pickBy, isEmpty } from 'lodash';
+import PropTypes from 'prop-types';
+import { __ } from '@eventespresso/i18n';
+
+/**
+ * WordPress dependencies
+ */
+import { Placeholder, SelectControl, Spinner } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+import { Component, Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { datetimeSelectOptions } from './options';
+
+const nowDateAndTime = moment();
+
+/**
+ * DatetimeSelect component.
+ * A react component for an datetime selector.
+ *
+ * @param {Array} datetimes  An empty array or array of Datetime Entities. See
+ *                          prop-types for shape.
+ * @param {function} onDatetimeSelect  The callback on selection of datetime.
+ * @param {string} selectLabel The label for the select input.
+ * @param {number} selectedDatetimeId  If provided, the id of the datetime to
+ *   pre-select.
+ * @param {number} forEventId   ID for Event to retrieve datetimes from
+ * @param {boolean} isLoading  Whether or not the selector should start in a
+ *                               loading state
+ * @return {Function}  A pure component function.
+ * @constructor
+ */
+class DatetimeSelect extends Component {
+	constructor() {
+		super( ...arguments );
+		this.placeHolder = this.placeHolder.bind( this );
```

Comment:
This should still be removed please.

Reply:
Also when you remove this, there will be no need for the constructor.

Reply:
that method uses `this` on its first line
```
	placeHolder() {
		const { isLoading, selectLabel } = this.props;
```

Reply:
I realize that.  But `placeHolder` is not passed through as a prop on an eventHandler so afaik that means it doesn't need `this` bound to that context.  You're simply calling it in the scope of the class its defined in.

Reply:
I also forgot to mention that I'd like to see some tests on this branch before merging.